build-tests/api-documenter-test/etc/yaml/api-documenter-test/docclass1.yml

```diff
@@ -20,16 +20,16 @@ items:
       - api-documenter-test.IDocInterface2
     package: api-documenter-test
     children:
-      - api-documenter-test.DocClass1.deprecatedExample
-      - api-documenter-test.DocClass1.exampleFunction
-      - api-documenter-test.DocClass1.exampleFunction_1
-      - api-documenter-test.DocClass1.interestingEdgeCases
+      - api-documenter-test.DocClass1.deprecatedExample()
+      - 'api-documenter-test.DocClass1.exampleFunction(string,string)'

```

Comment:
@rbuckton  What would you do with this input?  🙃
```ts
export declare function pathologicalExample(options: {
  args: {
    [name: string]: string | boolean;
  };
  buildErrorIconPath?: string;
  buildSuccessIconPath?: string;
  distFolder: string;
  gulp: GulpProxy | Gulp;
  isRedundantBuild?: boolean;
  jestEnabled?: boolean;
  libAMDFolder?: string;
  libES6Folder?: string;
  libESNextFolder?: string;
  libFolder: string;
  onTaskEnd?: (taskName: string, duration: number[], error?: any) => void;
  onTaskStart?: (taskName: string) => void;
  packageFolder: string;
  production: boolean;
  properties?: {
    [key: string]: any;
  };
  relogIssues?: boolean;
  rootPath: string;
  shouldWarningsFailBuild: boolean;
  showToast?: boolean;
  srcFolder: string;
  tempFolder: string;
  uniqueTasks?: IExecutable[];
  verbose: boolean;
} & Partial<ICopyConfig>): void;
```

Reply:
```yaml
items:
- uid: package.pathologicalExample
  name: package.pathologicalExample
  type: function
  syntax:
    parameters:
    - id: options
      type:
      - anonymous0:local
    return:
      type:
      - void
references:
- uid: anonymous0:local
  name: |> <content of options type>
  spec.typeScript:
  - name: '{\n  args: {\n    [name: string]: string | boolean;\n  }; <snip> gulp: '
  - name: GulpProxy
    uid: otherpackage.GulpProxy
  - name: ' | '
  - name: Gulp
    uid: gulp.Gulp
  - name: ';\n  isRedundantBuild?: <snip> uniqueTasks?: '
  - name: IExecutable
    uid: otherpackage.IExecutable
  - name: '[];\n  verbose: boolean;\n} & '
  - name: Partial
    uid: Partial
  - name: '<'
  - name: ICopyConfig
    uid: otherpackage.ICopyConfig
  - name: '>'
- uid: otherpackage.GulpProxy
  name: GulpProxy
- uid: gulp.Gulp
  name: Gulp
- uid: otherpackage.IExecutable
  name: IExecutable
- uid: otherpackage.ICopyConfig
  name: ICopyConfig
```

You also *could* have a uid of `{{args:{{[name:string]~string|boolean}},buildErrorIconPath?:string,buildSuccessIconPath?:string,...}&Partial{ICopyConfig}` (with the `...` replaced) since it doesn't need to be human readable, just machine readable.

Reply:
The entry in `references` is effectively the same as an `Excerpt`, with some additional information (`uid`, `nameWithType`, `fullName`).

Reply:
> it doesn't need to be human readable, just machine readable.

@rbuckton From a technical perspective, this may be true.  But I'm pretty uncomfortable about designing a system that emits huge unwieldy expressions in a file that people may have to understand/debug.

Also, what about the other related ID problems called out in https://github.com/microsoft/web-build-tools/issues/1308?  We wouldn't want these huge expressions to appear in:
- Markdown URLs
- DocFX URLs
- Titles that appear in member tables in the docs

Reply:
Complex types should probably have aliased UIDs (possibly based on a hash of the stable UID). They would point to an entry in the yaml `references` list which would have the expanded text/excerpt that breaks down the type into hyperlinkable tokens.